server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -129,6 +150,147 @@ public CompletionStage<Result> review(
     return editOrReview(request, applicantId, programId, blockId, true);
   }
 
+  /** This method handles the applicant's selection from the address correction options. */
+  @Secure
+  public CompletionStage<Result> confirmAddress(
+      Request request, long applicantId, long programId, String blockId) {
+    CompletionStage<Optional<String>> applicantStage = this.applicantService.getName(applicantId);
+
+    CompletableFuture<Void> applicantAuthCompletableFuture =
+        applicantStage
+            .thenComposeAsync(
+                v -> checkApplicantAuthorization(profileUtils, request, applicantId),
+                httpExecutionContext.current())
+            .toCompletableFuture();
+
+    CompletableFuture<ReadOnlyApplicantProgramService> applicantProgramServiceCompletableFuture =
+        applicantStage
+            .thenComposeAsync(
+                v -> applicantService.getReadOnlyApplicantProgramService(applicantId, programId),
+                httpExecutionContext.current())
+            .toCompletableFuture();
+
+    return CompletableFuture.allOf(
+            applicantAuthCompletableFuture, applicantProgramServiceCompletableFuture)
+        .thenApplyAsync(
+            (v) -> {
+              ReadOnlyApplicantProgramService roApplicantProgramService =
+                  applicantProgramServiceCompletableFuture.join();
+              Optional<Block> block = roApplicantProgramService.getBlock(blockId);
+              Optional<String> maybeAddressJson = request.session().get(ADDRESS_JSON_SESSION_KEY);
+
+              if (block.isEmpty()) {
+                return notFound();
+              }
+
+              if (maybeAddressJson.isEmpty()) {
```

Comment:
Question: How should this be tested? I always seem to meet this conditional when testing it. When I click "Save and Next" from the verify address screen, this error is thrown.


https://user-images.githubusercontent.com/4401157/220133767-2a90384d-0c6c-4db8-a357-9c427c5a2411.mov

Reply:
@seanmalbert  There are browser tests that check it. I tried using the address you have, but it works for me and I can't repro the error you got.

Reply:
@seanmalbert I think our fixes earlier should have this fixed.

Reply:
@gwendolyngoetz , yea! 🎉